src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -296,15 +298,14 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
      * Go to next submission
      */
     nextSubmission() {
-        this.programmingSubmissionService.getProgrammingSubmissionForExerciseForCorrectionRoundWithoutAssessment(this.exercise.id!, true).subscribe(
+        this.programmingSubmissionService.getProgrammingSubmissionForExerciseForCorrectionRoundWithoutAssessment(this.exercise.id!, true, this.correctionRound).subscribe(
             (response: ProgrammingSubmission) => {
                 const unassessedSubmission = response;
                 this.router.onSameUrlNavigation = 'reload';
                 // navigate to the new assessment page to trigger re-initialization of the components
-                this.router.navigateByUrl(
-                    `/course-management/${this.course!.id}/programming-exercises/${this.exercise.id}/code-editor/${unassessedSubmission.participation?.id}/assessment`,
-                    {},
-                );
+                let url = `/course-management/${this.course!.id}/programming-exercises/${this.exercise.id}/code-editor/${unassessedSubmission.participation?.id}/assessment`;
+                url += `?correction-round=${this.correctionRound}`;
+                this.router.navigateByUrl(url, {});
```

Comment:
I assume this wants to use `skipLocationChange` as well?

Reply:
Honestly I'm not sure, but as far I know and have tested we don't need it here. What would be the benefit of using it here? :D

Reply:
It means that you don't add state to the browser navigation (back, forward buttons), which makes sense here I think since you just add the `?...` to the url.